app/routes/metadata.rb

```diff
@@ -1,28 +1,30 @@
+require_relative '../../x'
+
 module ExercismWeb
   module Routes
     class Metadata < Core
-      get '/exercises/:track_id/:slug' do |track_id, slug|
-        status, response = Xapi.get("v2", "exercises", track_id, slug)
-        data = JSON.parse(response)
-        if status == 404
-          flash[:notice] = data['error']
+      get '/exercises/:id/:slug/tests' do |track_id, slug|
+        exercise = X::Exercise::TestFiles.find(track_id, slug)
+        if exercise.files.empty?
+          flash[:notice] = error_message(exercise)
           redirect '/'
```

Comment:
@kytrinyx We can have a special for that, where we explain the error and how to contribute.
What do you think ?